docs/diagrams/UndoSequenceDiagram-Model.puml

```diff
@@ -13,10 +13,15 @@ activate Model
 Model -> VersionedAddressBook : undo()
 activate VersionedAddressBook
 
-VersionedAddressBook -> VersionedAddressBook :resetData(ReadOnlyAddressBook)
-VersionedAddressBook --> Model :
+VersionedAddressBook --> Model : prevAddressBook
 deactivate VersionedAddressBook
 
+Model -> Model : setAddressBook()
+activate Model
+
+Model --> Model :
+deactivate Model
```

Comment:
Do we want to make this more detailed or leave it at this level of abstraction?

Reply:
Possibly, let's discuss with the group